sunpy/instr/goes.py

```diff
@@ -227,55 +224,55 @@ def calculate_temperature_em(goeslc, abundances="coronal",
+    >>> goests.data  # doctest: +REMOTE_DATA
+                                        xrsa          xrsb
+    2011-06-06 23:59:59.961999  1.000000e-09  1.887100e-07
+    2011-06-07 00:00:02.008999  1.000000e-09  1.834600e-07
+    2011-06-07 00:00:04.058999  1.000000e-09  1.860900e-07
+    2011-06-07 00:00:06.104999  1.000000e-09  1.808400e-07
+    2011-06-07 00:00:08.151999  1.000000e-09  1.860900e-07
+    ...
+
+    >>> goests_new = calculate_temperature_em(goests)  # doctest: +REMOTE_DATA
+    >>> goests_new.data  # doctest: +REMOTE_DATA
+                                        xrsa          xrsb  temperature            em
+    2011-06-06 23:59:59.961999  1.000000e-09  1.887100e-07     3.503510  2.190626e+48
+    2011-06-07 00:00:02.008999  1.000000e-09  1.834600e-07     3.534262  2.055847e+48
+    2011-06-07 00:00:04.058999  1.000000e-09  1.860900e-07     3.518700  2.122771e+48
+    2011-06-07 00:00:06.104999  1.000000e-09  1.808400e-07     3.550100  1.990333e+48
+    2011-06-07 00:00:08.151999  1.000000e-09  1.860900e-07     3.518700  2.122771e+48
+    ...
 
     """
     # Check that input argument is of correct type
-    if not isinstance(goeslc, lightcurve.LightCurve):
-        raise TypeError("goeslc must be a LightCurve object.")
+    if not isinstance(goests, timeseries.XRSTimeSeries):
+        raise TypeError("goests must be a XRSTimeSeries object")
     if not download_dir:
         download_dir = get_and_create_download_dir()
 
     # Find temperature and emission measure with _goes_chianti_tem
     temp, em = _goes_chianti_tem(
-        u.Quantity(goeslc.data.xrsb, unit=u.W/(u.m)**2),
-        u.Quantity(goeslc.data.xrsa, unit=u.W/(u.m)**2),
-        satellite=goeslc.meta["TELESCOP"].split()[1],
-        date=goeslc.data.index[0],
+        goests.quantity("xrsb"),
+        goests.quantity("xrsa"),
+        satellite=goests.meta.metas[0]["TELESCOP"].split()[1],
+        date=goests.data.index[0],
         abundances=abundances, download=download, download_dir=download_dir)
 
-    # Enter results into new version of GOES LightCurve Object
-    # Use copy.deepcopy for replicating meta and data so that input
-    # lightcurve is not altered.
-    lc_new = lightcurve.LightCurve(meta=copy.deepcopy(goeslc.meta),
-                                   data=copy.deepcopy(goeslc.data))
-    lc_new.data["temperature"] = temp.value
-    lc_new.data["em"] = em.value
+    lc_new = timeseries.XRSTimeSeries(meta=copy.deepcopy(goests.meta),
+                                      data=copy.deepcopy(goests.data),
+                                      units=copy.deepcopy(goests.units))
+    lc_new = lc_new.add_column("temperature", temp)
+    lc_new = lc_new.add_column("em", em)
 
     return lc_new
 
 
-@u.quantity_input(longflux=u.W/u.m/u.m, shortflux=u.W/u.m/u.m)
-def _goes_chianti_tem(longflux, shortflux, satellite=8,
+@u.quantity_input
+def _goes_chianti_tem(longflux: u.W/u.m/u.m, shortflux: u.W/u.m/u.m, satellite=8,
```

Comment:
This syntax is new to me.  Presumably it means ```longflux``` input must be a quantity with equivalent units to ```u.W/u.m/u.m```?

Reply:
indeed.